doc/api/cli.md

```diff
@@ -595,6 +595,31 @@ Path to the file used to store the persistent REPL history. The default path is
 `~/.node_repl_history`, which is overridden by this variable. Setting the value
 to an empty string (`''` or `' '`) disables persistent REPL history.
 
+### `NODE_UNHANDLED_REJECTION=state`
+<!-- YAML
+added: REPLACEME
+-->
+
+Setting this environment variable allows fine grained control over the behavior
```

Comment:
Does using an environment variable mean that any loaded module can mutate that environment variable and thus change the behavior of the process with regards to unhandled rejections?

Reply:
Yes. To be fair that's also true with `process.on('uncaughtException'` and `process.on('unhandledRejection'`.


The nice part about an environment variable is that it can be backported to older versions of Node that can also benefit from this behavior.

Reply:
We could read the variable when node bootstraps and cache its value (not saying we should, I don't know)

Reply:
Maybe we should at least validate it on startup and crash if an unsupported value is set

Reply:
@targos that validation is already implemented :-)

@misterdjules yes, the environment variable could be manipulated but I do not see this as an issue. However, I could include a flag as well to make sure that is handled. That way the environment variable would just be ignored / throws an error if not set to the same value as the flag.

I would not want to cache the variable on bootstrap because some code might indeed set the environment variable a bit later.

Reply:
I think Julien was just asking a question rather than requesting a behaviour change.

Reply:
From the perspective of operators of a Node.js platform running other developers' code, being able to turn on an "exit on unhandled rejections" command line flag is more reliable because we can be confident that that flag is not going to be turned off or changed by any code loaded by the runtime.

This is the model of the `--abort-on-uncaught-exception` flag, it seems that the unhandled rejection behavior could follow that model.

Reply:
I am fine adding the flag. The only reason I would rather not add it in this PR is that the PR becomes bigger and bigger and it becomes tedious for the reviewers.

@misterdjules are you fine if I add a TODO to add a flag as well where I already describe the behavior?

Reply:
IMO a flag instead of an environment varialbe would be better for this case. It's more consistent with other flags like `--abort-on-uncaught-exception` and `--throw-deprecation`, and users could still use `NODE_OPTIONS="--throw-on-unhandled-rejection"`.

(not saying `--throw-on-unhandled-rejection` should be the name of the flag, just using as an example)

Reply:
@BridgeAR 

> are you fine if I add a TODO to add a flag as well where I already describe the behavior?

What is the motivation for exposing this as an environment variable in the first place?